csharp/src/Google.Protobuf.Test.NativeAot/Program.cs

```diff
@@ -0,0 +1,67 @@
+﻿#region Copyright notice and license
+// Protocol Buffers - Google's data interchange format
+// Copyright 2016 Google Inc.  All rights reserved.
+// https://developers.google.com/protocol-buffers/
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//     * Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above
+// copyright notice, this list of conditions and the following disclaimer
+// in the documentation and/or other materials provided with the
+// distribution.
+//     * Neither the name of Google Inc. nor the names of its
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#endregion
+
+using Google.Protobuf.Test.NativeAot.Reflection;
+
+try
+{
+    // This project tests specific Google.Protobuf features with .NET Native AOT.
+    // It is designed to be published as a native app, and then run.
+    // Return value of 100 is success, 101 is failure.
```

Comment:
Any reason for returning 100 vs 101, instead of 0 vs 1? The latter would be simpler to handle in shell scripts etc, where 0 is normally "success" and anything else is failure.

(We could potentially have an `Action[]` and test each in turn, catching exceptions, and returning the number of failures.)

Sorry for not spotting this before.

Reply:
I followed some other tests which used 100 and 101. I've changed it to 0 and 1.

Using `Action[]` would be nice, but something to consider with AOT and trimming is running things in isolation. I think the ideal approach is adding `#ifdef` and then building and running the app for each test, passing in the test name as a compiler constant. I added that to the comment.